keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt

```diff
@@ -103,15 +99,47 @@ class ResourceStatusService(
   }
 
   /**
-   * Checks last 10 events for flapping between only ResourceActuationLaunched and ResourceDeltaDetected
+   * Returns true if a resource has been vetoed by the unhappy veto,
+   * or if the last 10 events are only ResourceActuationLaunched or ResourceDeltaDetected events,
+   * or if the resource has been vetoed by an unspecified veto that we don't have an explicit status mapping for.
    */
   private fun List<ResourceHistoryEvent>.isUnhappy(): Boolean {
+    if (first() is ResourceActuationVetoed && (first() as ResourceActuationVetoed).getStatus() == UNHAPPY) {
+      return true
+    }
+
     val recentSliceOfHistory = this.subList(0, Math.min(10, this.size))
     val filteredHistory = recentSliceOfHistory.filter { it is ResourceDeltaDetected || it is ResourceActuationLaunched }
-    if (filteredHistory.size != recentSliceOfHistory.size) {
-      // there are other events, we're not thrashing.
-      return false
+    if (filteredHistory.size == recentSliceOfHistory.size) {
+      return true
     }
-    return true
+    return false
   }
+
+  /**
+   * Determines if last event was a veto because of a missing dependency
+   */
+  private fun List<ResourceHistoryEvent>.isMissingDependency(): Boolean =
+    first() is ResourceActuationVetoed && (first() as ResourceActuationVetoed).getStatus() == MISSING_DEPENDENCY
+
```

Comment:
This function now looks at the new `suggestedStatus` field and helps figure out what status a vetoed resource should have.

Reply:
I like!